src/lib/utils/filters.ts

```diff
@@ -29,6 +30,59 @@ function filter<T>(
     return fzf.find(searchText).map((result) => result.item) as T[];
 }
 
+/**
+ * Checks if a given paper matches all active project paper filters.
+ *
+ * @remarks
+ * Only filter that have one or more selected values are applied, the remaining filters are ignored.
+ * A project paper must match all *active* filters.
+ * This means for the stage, publisher, year or decision that the paper's stage, ... must be
+ * in the list of selected values of the filters.
+ * Regarding the reviewers or selected criteria, at least one review or selected criterion must be
+ * in the list of selected values of the filters.
+ *
+ * @param paper - The paper to evaluate against the filters.
+ * @param filters - The filters to apply.
+ * @returns True if the paper matches all filters, false otherwise.
+ */
+function matchesFilters(paper: Project_Paper, filters: ProjectPaperFilter) {
+    const { stages, reviewers, publishers, years, decisions, criteria } = filters;
+
+    if (stages.length > 0 && !stages.includes(String(paper.stage))) {
+        return false;
+    }
+
+    if (
+        reviewers.length > 0 &&
+        !paper.reviews.some((review) => reviewers.includes(review.userId))
+    ) {
+        return false;
+    }
+
+    if (publishers.length > 0 && !publishers.includes(String(paper.paper?.publisher))) {
+        return false;
+    }
+
+    if (years.length > 0 && !years.includes(String(paper.paper?.year))) {
+        return false;
+    }
+
+    if (decisions.length > 0 && !decisions.includes(String(paper.decision))) {
+        return false;
+    }
```

Comment:
What are your thoughts on this refactoring?
```suggestion
    const matchesFilter = (filter: string[], value: string | bigint | number | undefined) =>
        filter.length === 0 || (value !== undefined && filter.includes(String(value)));

    if (!matchesFilter(stages, paper.stage)) {
        return false;
    }

    if (
        reviewers.length > 0 &&
        !paper.reviews.some((review) => reviewers.includes(review.userId))
    ) {
        return false;
    }

    if (!matchesFilter(publishers, paper.paper?.publisher)) {
        return false;
    }

    if (!matchesFilter(years, paper.paper?.year)) {
        return false;
    }

    if (!matchesFilter(decisions, paper.decision)) {
        return false;
    }
```

Reply:
I agree, your code looks a bit cleaner. I will refactor it.